src/ray/util/memory.cc

```diff
@@ -50,17 +59,16 @@ void parallel_memcopy(uint8_t *dst,
 
   // Start all threads first and handle leftovers while threads run.
   for (int i = 0; i < num_threads; i++) {
-    threadpool[i] = std::thread(
-        std::memcpy, dst + prefix + i * chunk_size, left + i * chunk_size, chunk_size);
+    futures[i] = std::async(std::launch::async, [dst, prefix, left, chunk_size, i]() {
```

Comment:
Here the launch policy *should* execute the task in a different thread:

https://en.cppreference.com/w/cpp/thread/launch

Reply:
Demo: https://replit.com/join/cpirsftcns-michaelhuang10